.github/workflows/pytest.yml

```diff
@@ -35,9 +35,28 @@ jobs:
           python -m pip install .[test]
 
       - name: Run tests
+        id: tests
         run: make test
+        continue-on-error: true
 
-      - name: Analysing the code with pylint
-        if: ${{ matrix.python-version }} == '3.10'
+      - name: Pylint analysis
+        id: pylint_analysis
         run: |
-          pylint --fail-under=9 $(git ls-files '*.py')
\ No newline at end of file
+          python -m pylint --fail-under=10 $(git ls-files '*.py')
+        continue-on-error: true
+
+      - name: mypy analysis
+        id: mypy_analysis
+        run: |
+          python -m mypy *.py finquant
```

Comment:
Any PR CI runs are now going to fail if mypy finds anything it is not happy about. Making sure that type hints are implemented correctly.